client/pages/OriginalDataPoint/components/ButtonBar/ButtonBar.tsx

```diff
@@ -1,37 +1,51 @@
+/* eslint-disable no-alert */
 import React from 'react'
 import { useHistory, useParams } from 'react-router'
 import { AssessmentName } from '@meta/assessment'
-// import { useAppDispatch } from '@client/store'
+import { useAppDispatch } from '@client/store'
 import { useCountryIso } from '@client/hooks'
 import { useTranslation } from 'react-i18next'
 import { BasePaths } from '@client/basePaths'
-// import { useOriginalDataPoint } from '@client/store/data/originalDataPoint'
+import {
+  OriginalDataPointActions,
+  useOriginalDataPoint,
+  useIsOriginalDataPointLoading,
+} from '@client/store/pages/originalDataPoint'
+import { useAssessment, useCycle } from '@client/store/assessment'
 
 type Props = {
   canEditData: boolean
 }
 
 const ButtonBar: React.FC<Props> = (props) => {
   const { canEditData } = props
-  // const originalDataPoint = useOriginalDataPoint()
+  const originalDataPoint = useOriginalDataPoint()
 
-  // const dispatch = useAppDispatch()
+  const dispatch = useAppDispatch()
   const history = useHistory()
   const { assessmentName, section } = useParams<{ assessmentName: AssessmentName; section: string }>()
   const { i18n } = useTranslation()
   const countryIso = useCountryIso()
-  const disabled = true // TODO: useIsAutoSaveSaving() || !originalDataPoint.id
+  const disabled = !originalDataPoint.id || useIsOriginalDataPointLoading()
```

Comment:
do we ever reach the point here where there's no odp ? 
If yes, then render this component only when ODP is loaded.

Also, what is loading? updating you mean ?